models/measurements.py

```diff
@@ -40,9 +47,257 @@ class Measurements(Base, AutoIDMixin, SpatiallyIndexed):
+        doc="PSF flux of the measurement. "
+    )
+
+    flux_psf_err = sa.Column(
+        sa.Float,
+        nullable=False,
+        doc="PSF flux error of the measurement. "
+    )
+
+    flux_apertures = sa.Column(
+        sa.ARRAY(sa.Float),
+        nullable=False,
+        doc="Aperture fluxes of the measurement. "
+    )
+
+    flux_apertures_err = sa.Column(
+        sa.ARRAY(sa.Float),
+        nullable=False,
+        doc="Aperture flux errors of the measurement. "
+    )
+
+    aper_radii = sa.Column(
+        sa.ARRAY(sa.Float),
+        nullable=False,
+        doc="Radii of the apertures used for calculating flux, in pixels. "
+    )
+
+    best_aperture = sa.Column(
+        sa.SMALLINT,
+        nullable=False,
+        default=-1,
+        doc="The index of the aperture that was chosen as the best aperture for this measurement. "
+            "Set to -1 to select the PSF flux instead of one of the apertures. "
+    )
+
+    mjd = association_proxy('cutouts', 'sources.image.mjd')
+
+    exp_time = association_proxy('cutouts', 'sources.image.exp_time')
+
+    filter = association_proxy('cutouts', 'sources.image.filter')
+
+    @property
+    def mag_psf(self):
+        return -2.5 * np.log10(self.flux_psf) + self.zp.zp  # what about aperture correction?
+
+    @property
+    def mag_psf_err(self):
+        return np.sqrt( (2.5 / np.log(10) * self.flux_psf_err / self.flux_psf) ** 2 + self.zp.dzp ** 2)
+
+    @property
+    def mag_apertures(self):
+        return [-2.5 * np.log10(f) + self.zp.zp + self.zp.aper_cors[i] for i, f in enumerate(self.flux_apertures)]
+
+    @property
+    def mag_apertures_err(self):
+        return [
+            np.sqrt( (2.5 / np.log(10) * ferr / f) ** 2 + self.zp.dzp ** 2)
+            for f, ferr in zip(self.flux_apertures, self.flux_apertures_err)
+        ]
+
+    @property
+    def magnitude(self):
+        if self.best_aperture == -1:
+            return self.mag_psf
+        return self.mag_apertures[self.best_aperture]
+
+    @property
+    def magnitude_err(self):
+        if self.best_aperture == -1:
+            return self.mag_psf_err
+        return self.mag_apertures_err[self.best_aperture]
+
+    @property
+    def lim_mag(self):
+        return self.cutouts.sources.image.new_image.lim_mag_estimate  # TODO: improve this when done with issue #143
+
+    @property
+    def zp(self):
+        return self.cutouts.sources.image.new_image.zp
+
+    @property
+    def fwhm_pixels(self):
+        return self.cutouts.sources.image.get_psf().fwhm_pixels
+
+    @property
+    def psf(self):
+        return self.cutouts.sources.image.get_psf().get_clip(x=self.cutouts.x, y=self.cutouts.y)
+
+    @property
+    def pixel_scale(self):
+        return self.cutouts.sources.image.new_image.wcs.get_pixel_scale()
+
+    background = sa.Column(
+        sa.Float,
+        nullable=False,
+        doc="Background of the measurement, from a local annulus. Given as counts per pixel. "
```

Comment:
Going back to my comment on measurements where we want to have the ability to do photometry without using an annulus, you could either make the background Null in that case, or make the background 0 and the background_err 0.  We'd have to think about whether setting both to 0 were enough, or if that might cause issues somewhere down the road.

Reply:
The way I see it is that it is good to have the background measurements, but you don't have to use it. 
If the image is properly subtracted, then it should measure close to zero anyway, and supposedly be less noisy than the aperture, so subtracting that number should not introduce a lot of noise (and you don't have to subtract it). 

Another thing to think about is the variance in the annulus is a good estimator for the flux error. 

Finally, it sometimes helps to have the background measurement on hand, to rule out bad data. For example if the object seems to brighten on an image or two, but also the background, then it is indicative of a false detection (e.g., a passing asteroid).

Reply:
Right, but we need the ability to not use it even for that.

Reply:
I'll make an issue so we can merge this one as it is now.